torch/_dynamo/backends/distributed.py

```diff
@@ -509,18 +518,24 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
                     if buckets[0].opcount_increased_to_capture_external_output == 0:
                         buckets[0].paramsize_before_opcount_increase = buckets[0].size
                     buckets[0].opcount_increased_to_capture_external_output += 1
-
             if node.op == "call_module":
-                target = gm.get_submodule(node.target)
-                for name, param in target.named_parameters():
-                    if param.requires_grad and not self._ignore_parameter(param):
-                        buckets[0].size += param.untyped_storage().nbytes()
-                        buckets[0].params.append(f"{node.target}_{name}")
-                        buckets[0].param_ids.append(id(param))
+                target_mod = gm.get_submodule(node.target)
+                if target_mod not in processed_modules:
+                    self.add_module_params_to_bucket(
+                        target_mod, buckets[0], processed_modules, node.target
+                    )
+            elif node.op == "call_method":
+                target_mod = gm.try_get_submodule(node.args[0].target)
```

Comment:
Thanks and I understand it better now - I'd vote for doing:
```
try:
    gm.get_submodule(node.args[0].target)
except AttributeError:
    pass
```
instead of adding the `try_get_submodule()` method to nn.Module. 

The original `try ... except ...` block in your previous version encompassed a few Python operations and AttributeError can come from any of them, so the error attribution was not easy to do. However in this new version we are adding more code to nn/modules/module.py which is also something we want to avoid (more code = more maintenance cost). Doing `try ... except ...` just on the `gm.get_submodule(node.args[0].target)` might be the right trade-off here I believe.